File: scenarios/command/dotnet/command_client/UnlockCommandClient.cs
using MQTTnet.Client;
using MQTTnet.Client.Extensions;
using proto_messages;

namespace command_client;

[RequestTopic("vehicles/{clientId}/command/{commandName}/request")]
[ResponseTopic("vehicles/{clientId}/command/{commandName}/response")]
public class UnlockCommandClient : CommandClient<UnlockRequest, UnlockResponse>
{
    public UnlockCommandClient(IMqttClient mqttClient)
        : base(mqttClient, "unlock", new ProtobufSerializer(UnlockResponse.Parser))
    {
    }
}

File: scenarios/command/ts/commandClient/src/unlockCommandClient.ts
import { MqttClient } from 'mqtt';
import { CommandClient } from '@mqttapplicationsamples/mqttjsclientextensions';
import {
    UnlockRequest,
    UnlockResponse,
    UnlockRequestSerializer,
    UnlockResponseSerializer
} from '@mqttapplicationsamples/protomessages';

export const UnlockCommand = 'unlock';
export const RequestTopicPattern = 'vehicles/{clientId}/command/{commandName}/request';
export const ResponseTopicPattern = 'vehicles/{clientId}/command/{commandName}/response';

export class UnlockCommandClient extends CommandClient<UnlockRequest, UnlockResponse>
{
    constructor(mqttClient: MqttClient) {
        super(mqttClient, RequestTopicPattern, ResponseTopicPattern, UnlockCommand, new UnlockRequestSerializer(), new UnlockResponseSerializer());
    }
}
